components/org.wso2.carbon.identity.password.expiry/src/main/java/org/wso2/carbon/identity/password/expiry/util/PasswordPolicyUtils.java

```diff
@@ -292,7 +295,95 @@ private static boolean isPasswordExpiredUnderDefaultPolicy(String tenantDomain,
             throws PostAuthenticationFailedException {
 
         if (skipIfNoApplicableRules) return false;
-        return lastPasswordUpdatedTime == null || daysDifference >= getPasswordExpiryInDays(tenantDomain);
+        return StringUtils.isBlank(lastPasswordUpdatedTime) || daysDifference >= getPasswordExpiryInDays(tenantDomain);
+    }
+
+    /**
+     * This method returns password expiry time for the given user.
+     *
+     * @param tenantDomain         The tenant domain.
+     * @param tenantAwareUsername  The tenant aware username.
+     * @param groupIds             The group IDs of the user.
+     * @param roleIds              The role IDs of the user.
+     * @return The password expiry time in milliseconds.
+     * @throws PostAuthenticationFailedException If an error occurred while getting the password expiry time.
+     */
+    public static Long getUserPasswordExpiryTime(String tenantDomain, String tenantAwareUsername,
```

Comment:
We can consider changing the return type from `Long` to `Optional<Long>` to better indicate that a `null` value is possible and make it clearer that a missing value is expected.

Reply:
Changed with [76bdf04](https://github.com/wso2-extensions/identity-governance/pull/856/commits/76bdf0463bf8cd623510402776938e7733c9feb6).